### PR TITLE
Use anyhow instead of boxed errors for parsers

### PR DESCRIPTION
**Pull request by kylewillmon**
We make heavy use of the anyhow crate everywhere else and it makes sense
to use it in the parsers as well.

This is just the first of a series of refactors to the parsers that I
hope to make in order to improve their ergonomics and assist in
implementing a `parse` subcommand (Ref: #346)

### CODE REVIEW

**Review by matt-phylum — APPROVED** [review]

File: cli/src/lockfiles/javascript.rs [hunk 1]
```diff
@@ -74,12 +75,14 @@ impl Parseable for YarnLock {
             Ok(yaml) => yaml,
             Err(_) => {
                 let (_, entries) =
-                    yarn::parse(&self.0).map_err(|_e| "Failed to parse yarn lock file")?;
+                    yarn::parse(&self.0).map_err(|_e| anyhow!("Failed to parse yarn lock file"))?;
```
[the remark this answers is not in the document]
Comment (cd-work) [reply cli/src/lockfiles/javascript.rs:78]: What does it look like when you just convert it to a string? Also the "failed to parse yarn lock file" should maybe be provided as context?

[the remark this answers is not in the document]
Comment (matt-phylum) [reply cli/src/lockfiles/javascript.rs:78]: It's funny. There is a to_owned and you can call it, but it just clones the errors vector without cloning the input strings. I started looking at making an `OwnedVerboseError` that you can do `map_err(OwnedVerboseError::from)`, but it's probably not worth it unless we're going to do something with that information besides printing it to the user later.

We probably want to use `convert_error`. https://github.com/Geal/nom/blob/02433c2c98a0448d30818ee057d303f2e63b8288/doc/error_management.md#getting-more-information-nomerrorverboseerror

[the remark this answers is not in the document]
Comment (kylewillmon) [reply cli/src/lockfiles/javascript.rs:78]: > What does it look like when you just convert it to a string? Also the "failed to parse yarn lock file" should maybe be provided as context?

It was surprisingly difficult to get this output.... But using this code:

```
                let (_, entries) = yarn::parse(&self.0)
                    .map_err(|e| anyhow!("{}", e))
                    .context("Failed to parse yarn lock file")?;
```

It looks like this:

```
[2022-05-05T15:30:53Z ERROR phylum] Execution failed: Unable to locate any valid package in package lockfile
    
    Caused by:
        0: Failed to parse yarn lock file
        1: Parsing Error: VerboseError { errors: [("", Nom(TakeUntil)), ("", Nom(Alt)), ("{\"a\": 100\n", Nom(Count))] }
```

[the remark this answers is not in the document]
Comment (kylewillmon) [reply cli/src/lockfiles/javascript.rs:78]: Alright. I think this is what we've been looking for. `nom` has a [`convert_error`](https://docs.rs/nom/7.1.1/nom/error/fn.convert_error.html#) function that adds line/column numbers to the `VerboseError` output. I wouldn't call it easy to read, but at least it points to specific locations:

```
[2022-05-05T16:14:44Z ERROR phylum] Execution failed: Unable to locate any valid package in package lockfile
    
    Caused by:
        0: Failed to parse yarn lock file
        1: 0: at line 2, in TakeUntil:
           
           ^
           
           1: at line 2, in Alt:
           
           ^
           
           2: at line 1, in Count:
           {"a": 100
           ^
```

**Pull request merged**